### PR TITLE
Do not use com.google.common.io.NullOutputStream

### PR DESCRIPTION
**Pull request by jglick**
Subsumes #230 and amends a mistake in #220: use of a beta API from the version of Guava currently bundled in Jenkins core, which has [subsequently been deleted](https://google.github.io/guava/releases/14.0/api/docs/com/google/common/io/NullOutputStream.html), causing mayhem if you try to run Pipeline-based tests with a newer Guava in the test classpath.

From [this search](https://github.com/search?q=user%3Ajenkinsci+%22com.google.common.io.NullOutputStream%22&type=Code) this seems to be the only usage in non-test code of this class except for `openshift-deployer` which has not been touched in three years.

### CODE REVIEW

File: src/main/java/org/jenkinsci/plugins/workflow/cps/Snippetizer.java [hunk 1]
```diff
@@ -535,7 +535,7 @@ public HttpResponse doGenerateSnippet(StaplerRequest req, @QueryParameter String
         return ExtensionList.lookup(SnippetizerLink.class);
     }
 
-    @Restricted(DoNotUse.class)
+    @Restricted(NoExternalUse.class)
     @Extension public static class PerJobAdder extends TransientActionFactory<Job> {
```
Comment (svanoort) [inline src/main/java/org/jenkinsci/plugins/workflow/cps/Snippetizer.java]: Why?

Comment (jglick) [reply src/main/java/org/jenkinsci/plugins/workflow/cps/Snippetizer.java]: Explained in #230.

File: src/main/java/org/jenkinsci/plugins/workflow/cps/actions/ArgumentsActionImpl.java [hunk 2]
```diff
@@ -276,8 +274,6 @@ Object sanitizeObjectAndRecordMutation(@CheckForNull Object o, @CheckForNull Env
         }
     }
 
-    static final OutputStream NULL_STRM = new NullOutputStream();
```
Comment (svanoort) [inline src/main/java/org/jenkinsci/plugins/workflow/cps/actions/ArgumentsActionImpl.java]: Why did you remove this?

File: src/main/java/org/jenkinsci/plugins/workflow/cps/actions/ArgumentsActionImpl.java [hunk 3]
```diff
@@ -25,19 +25,17 @@
 package org.jenkinsci.plugins.workflow.cps.actions;
 
 import com.google.common.collect.Maps;
-import com.google.common.io.NullOutputStream;
 import hudson.EnvVars;
 import jenkins.model.Jenkins;
+import org.apache.commons.io.output.NullOutputStream;
```
Comment (svanoort) [inline src/main/java/org/jenkinsci/plugins/workflow/cps/actions/ArgumentsActionImpl.java]: This is fine

File: src/main/java/org/jenkinsci/plugins/workflow/cps/actions/ArgumentsActionImpl.java (hunk 2 above)
Comment (jglick) [reply src/main/java/org/jenkinsci/plugins/workflow/cps/actions/ArgumentsActionImpl.java]: Because it is a flyweight object (no fields), so there is no need to cache it.

**Comment by svanoort** [comment]
pending build, then merge-ready

**Comment by jglick** [comment]
cf. https://github.com/jenkinsci/jenkins/pull/3481
